components/state.js

```diff
@@ -0,0 +1,26 @@
+// state.js
+
+var _ = require('underscore');
+
+var Lm = require('./Lm');
+
+var states = [];
+
+// Returns the vni associated with the specified vnid.
+module.exports = function( inportName, sourceNodeName, sourcePortName, newState ) { 
+ 
+   if ( arguments.length === 1 || arguments.length === 3 ) { 
+       if ( _.isUndefined( states[inportName] ) ) { 
+           return;
+       } 
+       return states[inportName];
+   }
+
+   if ( arguments.length === 2 || arguments.length === 4 ) { 
+       states[inportName] = { data: arguments[arguments.length-1],
+                              lm: Lm(),
+                              previousLms: [],
+                              port: inportName };
```

Comment:
Fair enough, but this is not really part of this sprint or writing a behavioural unit test which is what I am trying to get. I will stub the port as null for now to make it clear it is not set.